ch_backup/ch_backup.py

```diff
@@ -156,14 +156,17 @@ def backup(
 
         self._context.backup_layout.upload_backup_metadata(self._context.backup_meta)
 
+        skip_lock = self._check_schema_only_backup_skip_lock(sources)
+        logging.debug("Lock will be skipped for backup: {skip_lock}")
+
         logging.debug(
             'Starting backup "%s" for databases: %s',
             self._context.backup_meta.name,
             ", ".join(map(lambda db: db.name, databases)),
         )
 
         try:
-            with self._context.locker():
+            with self._context.locker(skip_lock=skip_lock):
```

Comment:
I would propose not to add parameter for locker. Why does one need a locker that does not lock anything ).
But do something like this:
```
if skip_lock:
    locker = contextlib.nullcontext()
    logging.debug("Lock will be skipped for backup")
else:
    locker = self._context.locker()

with locker:
        # Do something
```

Reply:
What is about changing parameter's name: `with self._context.locker(disabled=skip_lock):` ?

For me it seems more concisely - disabled locker makes some sense now.
And in case of necessity there will not be such if/else construction in other places.

Reply:
Okay, I don't mind. My option isn't dramatically better. But anyway log message should be conditional.